### PR TITLE
Fix `BaseFeePerGas` endianess

### PR DESCRIPTION
**Pull request by terencechain**
Fixed payload's `UnmarshalJSON` to use correct endianness on field `BaseFeePerGas`. Added more entropies to some tests

### CODE REVIEW

**Review by rauljordan — APPROVED** [review]

**Comment by rauljordan** [comment]
needs codeowner review

File: beacon-chain/state/v3/setters_test.go [hunk 1]
```diff
@@ -39,7 +39,7 @@ func TestAppendBeyondIndicesLimit(t *testing.T) {
 		ReceiptRoot:      make([]byte, 32),
 		LogsBloom:        make([]byte, 256),
 		PrevRandao:       make([]byte, 32),
-		BaseFeePerGas:    make([]byte, 32),
+		BaseFeePerGas:    bytesutil.PadTo([]byte{1, 2, 3, 4}, fieldparams.RootLength),
```
Comment (rkapka) [inline beacon-chain/state/v3/setters_test.go:42]: Please make the same changes in `beacon-chain/native-state/v3/setters_test.go`

**Review by rkapka — CHANGES_REQUESTED** [review]

File: beacon-chain/state/v3/setters_test.go [hunk 2]
```diff
@@ -123,7 +123,7 @@ func TestBeaconState_AppendBalanceWithTrie(t *testing.T) {
...
```
Comment (rkapka) [inline beacon-chain/state/v3/setters_test.go:126]: Also here

**Review by rkapka — APPROVED** [review]

**Review by nisdas — COMMENTED** [review]
A new test case should have been added in rather than modifying existing ones.

**Comment by terencechain** [comment]
> A new test case should have been added in rather than modifying existing ones.

The old test case is bogus though, and it would have resulted the same regardless of endianess

**Comment by nisdas** [comment]
> The old test case is bogus though, and it would have resulted the same regardless of endianess

The old test case doesn't seem to have been testing endianess specifically seems just like a basic mock. It would have been good to have one to test the endianess issue specifically.

**Comment by terencechain** [comment]
> > The old test case is bogus though, and it would have resulted the same regardless of endianess
> 
> The old test case doesn't seem to have been testing endianess specifically seems just like a basic mock. It would have been good to have one to test the endianess issue specifically.

Exactly. That's what was added: https://github.com/prysmaticlabs/prysm/pull/10376/files#diff-19fe51962300879516f288890ebfcb97ae51162dca32b0c5ccc2028a2d7b12fbR81

**Pull request merged**